File: merge sorted array.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        nums1size=len(nums1)
        while nums1size>m:
            nums1.pop()
            nums1size=nums1size-1

        for i in nums2:
            nums1.append(i)
        
        """now nums1 is merged and need to be sorted in ascending order!"""
        nums1.sort()
        print(nums1)

File: merge-sorted-array.java
//BRUTE FORCE BEATS 50% RUNTIME
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int i=m;
        int j=0;
        while(i<nums1.length && j<nums2.length){
            nums1[i]= nums2[j];
            i++;
            j++;
        }
        Arrays.sort(nums1);        
    }
}



//USING 2 POINTERS OPTIMAL SOLUTION
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
       int i=m-1;
       int j=n-1;
       int k=m+n-1;
       while(j>=0){
           if( i>=0 && nums1[i] > nums2[j] ) 
                nums1[k--]=nums1[i--];
            else
                nums1[k--]=nums2[j--];
       }
    }
}
